### PR TITLE
Add Database Iterations

### PR DESCRIPTION
**Pull request by jelilat**
## Description

This PR allows us to be able to query the database using prefixes. It will be useful when querying all the`Transactions` and `Receipts` in a `Block`. 

We will now be able to query all `Transactions` using the block number as a prefix.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Testing

**Requires testing**: Yes

**Did you write tests??**: Yes

### CODE REVIEW

**Comment by omerfirmak** [comment]
Yeap, this is what I had planned as well. Lets see if the original PR gets merged first.

**Comment by IronGauntlets** [comment]
hi @jelilat, can you please cherry-pick your `Add Db Iterator` on the latest commit on `develop`. Rebasing on develop will be much more difficult and reintroduce some code which was deleted.

File: db/db.go [hunk 1]
```diff
@@ -34,10 +34,41 @@ type Entry struct {
...
+
+	// Value returns the value at the current position.
+	Value() []byte
+
+	// Close closes the iterator.
+	Close() error
+
+	// Seek would seek to the provided key if present. If absent, it would seek to the next
+	// key in lexicographic order
+	Seek(cb func(*Entry) error) error
```
Comment (omerfirmak) [inline db/db.go:63]: How does seek work with no key input?

File: db/db.go [hunk 2]
```diff
@@ -34,10 +34,41 @@ type Entry struct {
 	Value []byte
 }
 
+type IterOptions struct {
```
Comment (omerfirmak) [inline db/db.go:37]: this is too pebble specific, should go.

Comment (omerfirmak) [reply db/db.go:37]: instead have a

Seek(key []byte) function

**Comment by jelilat** [comment]
Close in favour of https://github.com/NethermindEth/juno/pull/572

**Comment by omerfirmak** [comment]
You can force push to your branches to avoid creating new pull requests.

**Pull request closed without merging**